### ISSUE TITLE
Found with fuzzer: mon (61100095c6c0) at <5,7> is not there!

### ISSUE THREAD

**Issue by elunna** [elunna/hackem#17]
```

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
#2  0x0000555555a428ca in NH_abort () at end.c:236
#3  0x0000555555a496c6 in panic (str=0x55555651cb00 "%s") at end.c:802
#4  0x0000555555f9765b in impossible (s=0x5555564d50e0 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
#5  0x0000555555d62f17 in mon_sanity_check () at mon.c:164
#6  0x000055555585d3b3 in sanity_check () at cmd.c:4844
#7  0x0000555555762767 in moveloop (resuming=0 '\000') at allmain.c:767
#8  0x00005555563b5144 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737337632192, 93824997402012, 335544320, 
            140737341055120, 1073741824, 0, 140737342801568, 4294967295, 140737337360384, 
            140737342784672, 10981858626888671488}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7365859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342784672, 10981858626888671488, 
              140737488345360, 140737342801568, 18446744073709547520, 140737327112192, 
              140737345470680, 140737488346432, 140737488346432, 140737344339986, 
              17592186043170, 140737488338944, 140737341055120, 10981858626888671488, 
              140737488345760}}, sa_flags = -1656610560, sa_restorer = 0x7fffffffdc60}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a428ca in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a496c6 in panic (str=0x55555651cb00 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffdd30, 
            reg_save_area = 0x7fffffffdc60}}
#4  0x0000555555f9765b in impossible (s=0x5555564d50e0 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
        pbuf = "mon (61100095c6c0) at <5,7> is not there!\000\000\000\000\000\000\000\001\001\000\000\000\000\001\000\000\001\000\000\001\000\001\002\000\000\001\001\001\000\000\000\000\001", '\000' <repeats 39 times>, "\001\000\001\001\001\000\000\000\001\000\000\000\000\002\001\000\000\000\000\001", '\000' <repeats 68 times>...
        the_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7fffffffe420, 
            reg_save_area = 0x7fffffffe340}}
#5  0x0000555555d62f17 in mon_sanity_check () at mon.c:164
        x = 5
        y = 7
        mtmp = 0x61100095c6c0
        m = 0x611000950280
#6  0x000055555585d3b3 in sanity_check () at cmd.c:4844
No locals.
#7  0x0000555555762767 in moveloop (resuming=0 '\000') at allmain.c:767
        moveamt = 12
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 26662
        past_clock = 32495
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#8  0x00005555563b5144 in main (argc=0, argv=0x7fffffffe658) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
Found again:
```

Breakpoint 1, panic (str=0x3000000018 "") at end.c:730
730	VA_DECL(const char *, str)
#0  panic (str=0x3000000018 "") at end.c:730
#1  0x0000555555fa264c in impossible (s=0x5555564eb4e0 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
#2  0x0000555555d6b0fa in mon_sanity_check () at mon.c:165
#3  0x00005555558632c1 in sanity_check () at cmd.c:4858
#4  0x0000555555766ad3 in moveloop (resuming=0 '\000') at allmain.c:790
#5  0x00005555563ca0e4 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  panic (str=0x3000000018 "") at end.c:730
        the_args = {{gp_offset = 0, fp_offset = 0, overflow_arg_area = 0x0, 
            reg_save_area = 0x0}}
#1  0x0000555555fa264c in impossible (s=0x5555564eb4e0 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
        pbuf = "mon (6110005aa400) at <52,12> is not there!\000\000\000\000\000\001\001\000\000\000\000\001\000\000\001\000\000\001\000\001\002\000\000\001\001\001\000\000\000\000\001", '\000' <repeats 39 times>, "\001\000\001\001\001\000\000\000\001\000\000\000\000\002\001\000\000\000\000\001", '\000' <repeats 68 times>...
        the_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7fffffffe3d0, 
            reg_save_area = 0x7fffffffe2f0}}
#2  0x0000555555d6b0fa in mon_sanity_check () at mon.c:165
        x = 52
        y = 12
        mtmp = 0x6110005aa400
        m = 0x0
#3  0x00005555558632c1 in sanity_check () at cmd.c:4858
No locals.
#4  0x0000555555766ad3 in moveloop (resuming=0 '\000') at allmain.c:790
        moveamt = 10
        wtcap = 0
        change = 0
        monscanmove = 0 '\000'
        timeout_start = 30136
        past_clock = 597701
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#5  0x00005555563ca0e4 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
Popped up again:
```
Suddenly, the dungeon collapses.
mon (61100027fa00) at <49,9> is not there!
Generating more information you may report:

[0] /lib/x86_64-linux-gnu/libasan.so.5(+0x6cd40) [0x7ffff75f8d40]
[1] /home/lunatunez/games/hackemdir/hackem(+0x4fc9a6) [0x555555a509a6]
[2] /home/lunatunez/games/hackemdir/hackem(+0x4fc899) [0x555555a50899]
[3] /home/lunatunez/games/hackemdir/hackem(panic+0x6ab) [0x555555a576bb]
[4] /home/lunatunez/games/hackemdir/hackem(impossible+0x2c8) [0x555555fb2736]
[5] /home/lunatunez/games/hackemdir/hackem(mon_sanity_check+0x49c) [0x555555d77a48]
[6] /home/lunatunez/games/hackemdir/hackem(sanity_check+0x17) [0x555555867bd2]
[7] /home/lunatunez/games/hackemdir/hackem(moveloop+0x6c1c) [0x555555769842]
[8] /home/lunatunez/games/hackemdir/hackem(main+0xdda) [0x5555563e3115]
[9] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3) [0x7ffff7365083]
[10] /home/lunatunez/games/hackemdir/hackem(_start+0x2e) [0x555555760ace]

Program received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
#2  0x0000555555a508bf in NH_abort () at end.c:236
#3  0x0000555555a576bb in panic (str=0x55555654e0c0 "%s") at end.c:802
#4  0x0000555555fb2736 in impossible (s=0x555556505560 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
#5  0x0000555555d77a48 in mon_sanity_check () at mon.c:165
#6  0x0000555555867bd2 in sanity_check () at cmd.c:4857
#7  0x0000555555769842 in moveloop (resuming=0 '\000') at allmain.c:794
#8  0x00005555563e3115 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
        set = {__val = {0, 0, 0, 0, 0, 140737335783280, 93824997459345, 335544320, 
            140737341046928, 1073741824, 0, 140737342793376, 4294967295, 140737337360384, 
            140737342776480, 12408039515807777024}}
        pid = <optimized out>
        tid = <optimized out>
        ret = <optimized out>
#1  0x00007ffff7363859 in __GI_abort () at abort.c:79
        save_stage = 1
        act = {__sigaction_handler = {sa_handler = 0xffffffff, sa_sigaction = 0xffffffff}, 
          sa_mask = {__val = {140737337360384, 140737342776480, 12408039515807777024, 
              140737488345296, 140737342793376, 18446744073709547520, 140737327104000, 
              140737345462488, 140737488346368, 140737488346368, 140737344331794, 
              17592186043162, 140737488338944, 140737341046928, 12408039515807777024, 
              140737488345696}}, sa_flags = -222592768, sa_restorer = 0x7fffffffdc20}
        sigs = {__val = {32, 0 <repeats 15 times>}}
#2  0x0000555555a508bf in NH_abort () at end.c:236
        gdb_prio = 1
        libc_prio = 2
        aborting = 1 '\001'
#3  0x0000555555a576bb in panic (str=0x55555654e0c0 "%s") at end.c:802
        the_args = {{gp_offset = 16, fp_offset = 48, overflow_arg_area = 0x7fffffffdcf0, 
            reg_save_area = 0x7fffffffdc20}}
#4  0x0000555555fb2736 in impossible (s=0x555556505560 "mon (%s) at <%d,%d> is not there!")
    at pline.c:518
        pbuf = "mon (61100027fa00) at <49,9> is not there!\000VUU\000\000\000\000\000\000\000\000\000\001\000\000\000\000\000\000\000\000`\336\377\377\377\177\000\000&\323\346UUU\000\000\300\225\t\000\060a\000\000\360\335\377\377\377\177\000\000\020\337\377\377\377\177\000\000\\]\033VUU\000\000\n\000\000\000 \000\000\000\300\344'\000\020a\000\000\020\336\377\377\377\177\000\000\000\000\000\000\000\000\000\000`\336\377\377\377\177\000\000\vG\217U U\000\000\024x}VUU\000\000ʧ~VUU\000\000\300\351'\000\020a\000\000\200\350'\000\020a\000\000X=cVUU\000\000X"...
        the_args = {{gp_offset = 32, fp_offset = 48, overflow_arg_area = 0x7fffffffe3e0, 
            reg_save_area = 0x7fffffffe300}}
#5  0x0000555555d77a48 in mon_sanity_check () at mon.c:165
        x = 49
        y = 9
        mtmp = 0x61100027fa00
        m = 0x0
#6  0x0000555555867bd2 in sanity_check () at cmd.c:4857
No locals.
#7  0x0000555555769842 in moveloop (resuming=0 '\000') at allmain.c:794
        moveamt = 18
        wtcap = 0
        change = 0
        monscanmove = 1 '\001'
        elf_regen = 1 '\001'
        orc_regen = 1 '\001'
        vamp_regen = 1 '\001'
#8  0x00005555563e3115 in main (argc=0, argv=0x7fffffffe628) at ../sys/unix/unixmain.c:353
        fd = -1
        dir = 0x0
        exact_username = 0 '\000'
        resuming = 0 '\000'
        plsel_once = 1 '\001'
```

**Comment by elunna** [comment]
In the latest hit, this is for a bull. This appears to be related to steeds.

**Comment by elunna** [comment]
Using rr to trace this back a bit: I found one troll that was an enormous rat that stepped on a polytrap.
The problem seems to be with steeds stepping on polytraps - possibly being teleported or level teleported.

**Comment by elunna** [comment]
Possibly fixed by removing monster steeds in commit da87ef141.
